src/Mapster/Adapters/CollectionAdapter.cs

```diff
@@ -51,47 +46,62 @@ protected override bool CanInline(Expression source, Expression destination, Com
                 throw new InvalidOperationException($"{arg.DestinationType} is not supported for projection, please consider using List<>");
             }
 
-            if (arg.DestinationType == typeof (IEnumerable) || arg.DestinationType.IsGenericEnumerableType())
+            if (arg.DestinationType == typeof(IEnumerable) || arg.DestinationType.IsGenericEnumerableType())
                 return true;
 
             return false;
         }
 
         protected override Expression CreateInstantiationExpression(Expression source, Expression destination, CompileArgument arg)
         {
-            var destinationElementType = arg.DestinationType.ExtractCollectionType();
-            if (arg.DestinationType.IsArray)
-                return Expression.NewArrayBounds(
-                    destinationElementType, 
-                    CreateCountExpression(source, true));   //new TDestinationElement[count]
+            if (arg.SourceType.IsArray && arg.DestinationType.IsArray)
+            {
+                return CreateArrayInstantiationExpression(source, arg);
+            }
```

Comment:
Will this break change? I haven't test yet but seemed like we cannot map non-array to array any more.